### PR TITLE
Switch from Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by alexvanyo**
This PR switches from Travis CI to Github Actions for running CI.

With the use of `matrix`, I'm also taking this opportunity to start testing on JDK versions 1.8 through 14 (with 15 blocked by an update to JaCoCo).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      matrix:
+        java-version:
+          - 1.8
+          - 9
+          - 10
+          - 11
+          - 12
+          - 13
+          - 14
+        # - 15 TODO: Include JDK 15 when we can update JaCoCo to 0.8.7
```
Comment (dperez37) [inline .github/workflows/ci.yml:22]: What is blocking JaCoCo 0.8.7?

Comment (alexvanyo) [reply .github/workflows/ci.yml:22]: https://www.jacoco.org/jacoco/trunk/doc/changes.html

`0.8.7` has yet to be released, although there are snapshots available and it [should be soon™️](https://github.com/jacoco/jacoco/milestone/31). It seemed easiest to just hold off on running on JDK 15 until `0.8.7` is released.

**Review by dperez37 — COMMENTED** [review]
This stuff all seems fine.

Is there anything here that is hyper-specific to the project?

**Comment by alexvanyo** [comment]
Nope, nothing super specific! Jitpack handles publishing, so all CI needs to do is run the tests, and call Danger if it can.

For security purposes, Danger will only work right now for PRs originating from inside the project, and not from outside PRs.

That should be possible, but will require some more delicate setup to avoid security risks:
https://securitylab.github.com/research/github-actions-preventing-pwn-requests

**Review by dperez37 — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      - name: Configure JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java-version }}
+          
+      - name: Test
+        run: ./gradlew check jacocoTestReport
+
+      - name: Danger
+        if: github.event_name == 'pull_request' && matrix.java-version == '1.8' && github.event.pull_request.head.repo.full_name == github.repository
```
Comment (shadyoak) [inline .github/workflows/ci.yml:66]: Might be worth putting a comment here. Something to the effect of "ensures Danger is only run from PRs that originate from within the repo for security purposes" or similar. I like this a lot, though. Simple and effective.

**Review by shadyoak — APPROVED** [review]
Yup! 👍

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alexvanyo) [reply .github/workflows/ci.yml:67]: Updated!

**Comment by alexvanyo** [comment]
Thanks @dperez37 and @shadyoak !

**Pull request merged**